src/Illuminate/Database/Eloquent/Factories/HasFactory.php

```diff
@@ -23,6 +24,24 @@ public static function factory($count = null, $state = [])
                     ->state(is_callable($count) || is_array($count) ? $count : $state);
     }
 
+    /**
+     * Get the factory from the UseFactory class attribute.
+     *
+     * @return TFactory|null
+     */
+    protected static function getUseFactoryAttribute()
+    {
+        $attributes = (new \ReflectionClass(static::class))
+            ->getAttributes(UseFactory::class);
+
+        if (!empty($attributes)) {
```

Comment:
You could do this without a function call:
```suggestion
        if ($attributes !== []) {
```

Reply:
This is so clever, great suggestion.

Reply:
Thanks 😅 I thought the same thing when I first saw this in someone else's code review ^^